components/infobox/commons/infobox_player.lua

```diff
@@ -0,0 +1,227 @@
+            localizedname = args.name,
+            nationality = args.country or args.nationality,
+            nationality2 = args.country2 or args.nationality2,
+            nationality3 = args.country3 or args.nationality3,
+            birthdate = birthday,
+            deathdate = deathday,
+            image = args.image,
+            region = args.region,
+            team = args.teamlink or args.team,
+            status = status,
+            type = playerType,
+            earnings = earnings,
+            links = mw.ext.LiquipediaDB.lpdb_create_json(links),
+            extradata = mw.ext.LiquipediaDB.lpdb_create_json(extradata),
+        })
+    end
+
+    return infobox:build()
+end
+
+--- Allows for overriding this functionality
+function Player:getType(args)
+    return args.role
+end
+
+--- Allows for overriding this functionality
+function Player:getStatus(args)
+    return args.status
+end
+
+--- Allows for overriding this functionality
+function Player:getExtradata(args)
+    return {}
+end
+
+--- Allows for overriding this functionality
+--- Decides if we store in LPDB and Vars or not
+function Player:storeInLPDBandVars(args)
+    return Namespace.isMain()
+end
+
+--- Allows for overriding this functionality
+--- e.g. to add faction icons to the display for SC2, SC, WC
+function Player:name(args)
+    return args.id
+end
+
+--- Allows for overriding this functionality
+function Player:addCustomContent(infobox, args)
+    return infobox
+end
+
+--- Allows for overriding this functionality
+function Player:addCustomContent(infobox, args)
+    return infobox
+end
+
+--- Allows for overriding this functionality
+function Player:getAchievements(infobox, args)
+    return args.achievements
+end
+
+--- Allows for overriding this functionality
+function Player:addCustomCells(infobox, args)
+    return infobox
+end
+
+--- Allows for overriding this functionality
+function Player:calculateEarnings(args)
+    return error('You have not implemented a custom earnings function for your wiki')
+end
+
+--- Allows for overriding this functionality
+function Player:createBottomContent(infobox)
+    return infobox
+end
+
+function Player:_createRegion(region)
+    if region == nil or region == '' then
+        return ''
+    end
+
+    return Template.safeExpand(self.frame, 'Region', {region})
+end
+
+function Player:_createLocation(country, location)
+    if country == nil or country == '' then
+        return ''
+    end
+    local countryDisplay = Flags._CountryName(country)
+
+    return Flags._Flag(country) .. '&nbsp;' ..
+                '[[:Category:' .. countryDisplay .. ' Players|' .. countryDisplay .. ']]'
+                .. (location ~= nil and (',&nbsp;' .. location) or '')
+end
+
+function Player:_createTeam(team, link)
+    if team == nil or team == '' then
+        return ''
+    end
+    link = link or team
+
+    return '[[' .. link .. '|' .. team .. ']]'
+end
+
+--- Allows for overriding this functionality
```

Comment:
In what cases would this need an override?

Reply:
it doesn't i just forgot to remove the comment